File: 프로그래머스/1/140108. 문자열 나누기/문자열 나누기.py
def solution(s):
    cnt = 0
    x = []
    not_x = []
    for c in s:
        if not x:
            x.append(c)
        else:
            if c == x[0]:
                x.append(c)
            else:
                not_x.append(c)
        
        if(len(x) == len(not_x)):
            cnt += 1
            x = []
            not_x = []
            
    if x or not_x:
        cnt += 1
    
    return cnt

File: 프로그래머스/1/140108. 문자열 나누기/문자열 나누기.js
function solution(s) {
    let answer = 0;
    let q = s.split("");
    
    while(q.length > 0){
        let a = 1, b = 0;
        let x = q.shift();
        
        while(q.length > 0){
            let n = q.shift();
            
            if(n === x) a++;
            else b++;
            
            if(a === b){
                answer++;
                break;
            }
        }
        if(a !== b && q.length === 0) answer++;
    }
    return answer;
    
}